crates/bevy_math/src/rotation2d.rs

```diff
@@ -228,11 +227,10 @@ impl Rotation2d {
 
     /// Returns whether `self` has a length of `1.0` or not.
     ///
-    /// Uses a precision threshold of `1e-6`.
+    /// Uses a precision threshold of approximately `1e-4`.
     #[inline]
     pub fn is_normalized(self) -> bool {
-        let length = self.sin.hypot(self.cos);
-        length - 1.0 <= 1e-6
+        (self.length_squared() - 1.0).abs() <= 2e-4
```

Comment:
Maybe a comment explaining why we use `2e-4` in error analysis might be good for future reference.